posthog/tasks/calculate_event_property_usage.py

```diff
@@ -10,12 +11,34 @@
 from posthog.models import EventDefinition, EventProperty, Insight, PropertyDefinition, Team
 from posthog.models.filters.filter import Filter
 from posthog.models.property_definition import PropertyType
+from posthog.redis import get_client
+
+CALCULATED_PROPERTIES_FOR_TEAMS_KEY = "CALCULATED_PROPERTIES_FOR_TEAMS_KEY"
 
 
 @timed("calculate_event_property_usage")
 def calculate_event_property_usage() -> None:
+    teams_to_exclude = recently_calculated_teams(now_in_seconds_since_epoch=time.time())
+
     for team_id in Team.objects.values_list("id", flat=True):
-        calculate_event_property_usage_for_team(team_id=team_id)
+        if team_id not in teams_to_exclude:
+            calculate_event_property_usage_for_team(team_id=team_id)
+            get_client().zadd(name=CALCULATED_PROPERTIES_FOR_TEAMS_KEY, mapping={str(team_id): time.time()})
+
+
+def recently_calculated_teams(now_in_seconds_since_epoch: float) -> Set[int]:
+    """
+    Each time a team has properties calculated it is added to the sorted set with the seconds since epoch as its score.
+    That means we can read all teams in that set whose score is within the seconds since epoch covered in the last 24 hours
+    And exclude them from recalculation
+    """
+    one_day_ago = now_in_seconds_since_epoch - 86400
+    return {
+        int(team_id)
+        for team_id, _ in get_client().zrangebyscore(
```

Comment:
`ZRANGEBYSCORE` is deprecated:

https://redis.io/commands/zrangebyscore/

Reply:
🤔 hmmm, deprecated in redis but not in the python client. I wonder if their plan is to hide the deprecation behind the same method...

Reply:
Good catch though! Am using zrange in two places now so very interested to see what the what from that issue

Reply:
Hmm I'd have rather we did the suggested thing from the Redis docs here:

> It can be replaced by [ZRANGE](https://redis.io/commands/zrange) with the BYSCORE argument when migrating or writing new code.

Keeping in sync with Redis will pay off here even if the Python lib is behind

Reply:
Ugh, we're miles behind in client version 😵‍💫 

OK, can sort that

Reply:
(we don't have `byscore` and I only looked locally)